tests/pykafka/test_producer.py

```diff
@@ -20,12 +21,19 @@ def tearDownClass(cls):
 
     def test_produce(self):
         try:
-            prod = self.client.topics[self.topic_name].get_producer(batch_size=5)
-            prod.produce(['msg test'])
-
-            consumer = self.client.topics[self.topic_name].get_balanced_consumer('test_consume', zookeeper_connect=self.kafka.zookeeper)
-            messages = [consumer.consume() for i in xrange(1)]
-            self.assertTrue(len(messages) == 1)
+            # unique bytes, just to be absolutely sure we're not fetching data
+            # produced in a previous test
+            payload = uuid4().bytes
+
+            prod = self.client.topics[self.topic_name].get_producer(
+                batch_size=5)
+            prod.produce([payload])
+
+            # set a timeout so we don't wait forever if we break producer code
+            consumer = self.client.topics[self.topic_name].get_simple_consumer(
+                consumer_timeout_ms=1000)
```

Comment:
Good thing I just fixed the bug that makes this work the way it's supposed to! https://github.com/Parsely/pykafka/commit/3ffa301926825a86081a5603b4945a6bbc90f58e